docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc

```diff
@@ -1346,6 +1346,53 @@ Authentication that requires a dynamic tenant will fail.
 You can filter OIDC requests made by Quarkus to the OIDC provider by registering one or more `OidcRequestFilter` implementations, which can update or add new request headers, and log requests.
 For more information, see xref:security-oidc-code-flow-authentication#code-flow-oidc-request-filters[OIDC request filters].
 
+[[bearer-token-oidc-response-filters]]
+=== OIDC response filters
+
+You can filter responses from the OIDC providers by registering one or more `OidcResponseFilter` implementations, which can check the response status, headers and body in order to log them or perform other actions.
+
+You can have a single filter intercepting all the OIDC responses, or use an `@OidcEndpoint` annotation to apply this filter to the specific endpoint responses only. For example:
+
+[source,java]
+----
+package io.quarkus.it.keycloak;
+
+import jakarta.enterprise.context.ApplicationScoped;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.arc.Unremovable;
+import io.quarkus.oidc.common.OidcEndpoint;
+import io.quarkus.oidc.common.OidcEndpoint.Type;
+import io.quarkus.oidc.common.OidcResponseFilter;
+import io.quarkus.oidc.common.runtime.OidcConstants;
+import io.quarkus.oidc.runtime.OidcUtils;
+
+@ApplicationScoped
+@Unremovable
```

Comment:
Is `@Unremovable` always required? Wonder if beans with `@OidcEndpoint` can be considered `@Unremovable` by default

Reply:
@gastaldi Thanks, interesting idea, we can probably make an enhancement request to introspect it at build time. Though `@OidcEndpont` is optional, without it a filter applies to all endpoint  responses (in case of response filters)

Reply:
Right, or maybe consider `OidcResponseFilter` implementations unremovable?

Reply:
Yeah, for all 3 types of filters (request, response, redirect)